src/app/modules/courses/components/course-add-edit/course-add-edit.component.ts

```diff
@@ -17,75 +18,60 @@ import { Observable } from 'rxjs';
 export class CourseAddEditComponent implements OnInit {
 
   form = this.fb.group({
+    id: [],
     name: [],
     description: [],
     length: [],
     date: [],
     authors: []
   });
 
-  isNew = true;
+  authorsList$ = this.store.select(selectAuthors)
 
-  authorsList: Author[] = [];
-
-  private  currentCourseId: number;
+  currentCourseId = this.getId() || null;
 
   constructor(
     private fb: FormBuilder,
     private route: ActivatedRoute,
-    private coursesService: CoursesService,
-    private cdr: ChangeDetectorRef,
-    private router: Router
+    private router: Router,
+    private store: Store
   ) {
   }
 
   ngOnInit(): void {
-    this.coursesService
-      .getAuthors()
-      .pipe(untilDestroyed(this))
-      .subscribe(res => {
-        this.authorsList = res;
-        if (this.getId() !== undefined) {
-          this.seedForm(Number(this.getId()));
-        }
-        this.cdr.markForCheck();
-      });
+    this.store.dispatch(CoursesActions.enterToAddEditCoursePage({ courseId: this.currentCourseId }));
+    this.store.select(selectSelectedCourse).pipe(
+      filter(Boolean),
+      tap(course => this.seedForm(course)),
+      untilDestroyed(this)
+    ).subscribe()
   }
 
   private getId(): string | undefined {
     const { id } = this.route.snapshot.params;
     return id;
   }
 
-  private seedForm(id: number): void {
-    this.coursesService.getCourse(id).subscribe(res => {
-      this.form.setValue({
-        name: res.name,
-        description: res.description,
-        length: res.length,
-        date: new Date(res.date),
-        authors: this.authorsList.filter(author => res.authors.findIndex(x => x.id === author.id) !== -1)
-      });
-      this.currentCourseId = res.id;
-      this.isNew = false;
-      this.cdr.markForCheck();
+  private seedForm(course: Course): void {
+    this.form.setValue({
+      id: course.id,
+      name: course.name,
+      description: course.description,
+      length: course.length,
+      date: new Date(course.date),
+      authors: course.authors
     });
   }
 
   save(): void {
-    const obs$ = this.isNew ? this.saveNew() : this.saveExist();
-    obs$.subscribe(() => this.router.navigate(['/courses']));
-  }
-
-  saveNew(): Observable<unknown> {
-    return this.coursesService.createCourse(this.form.value as Course)
-  }
-
-  saveExist(): Observable<unknown> {
-    return this.coursesService.updateCourse(this.currentCourseId, this.form.value as Course);
+    const course = this.form.value as Course;
+    this.currentCourseId === null ?
+      this.store.dispatch(CoursesActions.createCourse({ course }))
+      : this.store.dispatch(CoursesActions.updateCourse({ course }));
+    this.navigateToMainPage();
```

Comment:
Here you are making navigate without waiting BE response. In case of BE error you should not redirect user, but should show error msg.